src/libstd/fs.rs

```diff
@@ -2261,11 +2282,41 @@ mod tests {
     }
 
     #[test]
+    fn concurrent_recursive_mkdir() {
+        for _ in 0..50 {
```

Comment:
It's because it relies on there being a race condition between the threads, but race conditions don't occur very often. By doing the test 50 times, it is much more likely to hit the race condition.